lua/bufferline/highlight.lua

```diff
@@ -0,0 +1,95 @@
+-- !::exe [So]
+
+-------------------
+-- Section: helpers
+-------------------
+
+--- Generate a color.
+--- @param index string where to look for the color.
+--- @param groups table<string> the groups to source the color from.
+--- @param default string the foreground color to use if no `groups` have a valid color.
+--- @return string color
+local function color(index, groups, default)
+  for _, group in ipairs(groups) do
+    local hl = vim.api.nvim_get_hl_by_name(group, true)
```

Comment:
I can probably sneak in #207 here by passing `vim.go.termguicolors` instead of `true`.